File: Sources 4.5.2/Gizmox.WebGUI.Forms.CompanionKit/Controls/GoogleMap/Features/MapControlTypePropertyPage.vb
Imports Gizmox.WebGUI.Forms.Google

Namespace CompanionKit.Controls.GoogleMap.Features

    Public Class MapControlTypePropertyPage

        Public Sub New()

            ' This call is required by the Visual WebGui UserControl Designer.
            InitializeComponent()

        End Sub

        ''' <summary>
        ''' Handles SelectedIndexChanged event for ComboBox
        ''' </summary>
        Private Sub mobjMapControlTypeComboBox_SelectedIndexChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles mobjMapControlTypeComboBox.SelectedIndexChanged
            ' Set map control type for GoogleMap control
            If (mobjMapControlTypeComboBox.SelectedIndex = 0) Then
                mobjGoogleMap.MapControlType = GoogleMapControlType.[Default]
            Else
                If (mobjMapControlTypeComboBox.SelectedIndex = 1) Then
                    mobjGoogleMap.MapControlType = GoogleMapControlType.DropdownMenu
                Else
                    mobjGoogleMap.MapControlType = GoogleMapControlType.HorizontalBar
                End If
            End If
        End Sub

        Private Sub mobjMapZoomControlTypeComboBox_SelectedIndexChanged(ByVal sender As Object, ByVal e As EventArgs) Handles mobjMapZoomControlTypeComboBox.SelectedIndexChanged
            ' Set map control type for GoogleMap control
            If (mobjMapZoomControlTypeComboBox.SelectedIndex = 0) Then
                mobjGoogleMap.MapZoomControlType = GoogleMapZoomControlType.[Default]
            Else
                If (mobjMapZoomControlTypeComboBox.SelectedIndex = 1) Then
                    mobjGoogleMap.MapZoomControlType = GoogleMapZoomControlType.Large
                Else
                    mobjGoogleMap.MapZoomControlType = GoogleMapZoomControlType.Small
                End If
            End If
        End Sub

    End Class

End Namespace

File: Sources 4.5.2/Gizmox.WebGUI.Forms.CompanionKit/Controls/GoogleMap/Features/MapControlTypePropertyPage.cs
using System;
using System.Text;
using System.Data;
using System.Drawing;
using System.ComponentModel;
using System.Collections.Generic;

using Gizmox.WebGUI.Forms;
using Gizmox.WebGUI.Common;
using Gizmox.WebGUI.Forms.Google;

namespace CompanionKit.Controls.GoogleMap.Features
{
    public partial class MapControlTypePropertyPage : UserControl
    {
        public MapControlTypePropertyPage()
        {
            InitializeComponent();
        }

        /// <summary>
        /// Handles SelectedIndexChanged event for ComboBox
        /// </summary>
        private void mobjMapControlTypeComboBox_SelectedIndexChanged(object sender, EventArgs e)
        {
            // Set map control type for GoogleMap control
            mobjGoogleMap.MapControlType = mobjMapControlTypeComboBox.SelectedIndex == 0 ? GoogleMapControlType.Default : (mobjMapControlTypeComboBox.SelectedIndex == 1 ? GoogleMapControlType.DropdownMenu : GoogleMapControlType.HorizontalBar);
        }

        /// <summary>
        /// Handles SelectedIndexChanged event for ComboBox
        /// </summary>
        private void mobjMapZoomControlTypeComboBox_SelectedIndexChanged(object sender, EventArgs e)
        {
            // Set map control type for GoogleMap control
            mobjGoogleMap.MapZoomControlType = mobjMapZoomControlTypeComboBox.SelectedIndex == 0 ? GoogleMapZoomControlType.Default : (mobjMapZoomControlTypeComboBox.SelectedIndex == 1 ? GoogleMapZoomControlType.Large : GoogleMapZoomControlType.Small);
        }

    }
}